packages/example-log-extension/test/unit/providers/log-action.provider.unit.ts

```diff
@@ -52,32 +53,64 @@ describe('LogActionProvider (unit)', () => {
     await logger(req, ['test', 'message'], 'test message');
     sinon.assert.calledWith(spy, match);
   });
+});
 
-  async function getLogger() {
-    class TestClass {
-      @log(LOG_LEVEL.ERROR)
-      test() {}
-    }
-
-    const context: Context = new Context();
-    context.bind(CoreBindings.CONTROLLER_CLASS).to(TestClass);
-    context.bind(CoreBindings.CONTROLLER_METHOD_NAME).to('test');
-    context.bind(EXAMPLE_LOG_BINDINGS.APP_LOG_LEVEL).to(LOG_LEVEL.WARN);
-    context.bind(EXAMPLE_LOG_BINDINGS.TIMER).to(timer);
-    context.bind(EXAMPLE_LOG_BINDINGS.LOG_ACTION).toProvider(LogActionProvider);
-    logger = await context.get(EXAMPLE_LOG_BINDINGS.LOG_ACTION);
-  }
+describe('LogActionProvider with default logger', () => {
+  let stub: sinon.SinonSpy;
+  let logger: LogFn;
+  const req = <ParsedRequest>{url: '/test'};
 
-  function createLogSpy() {
-    spy = sinon.spy(LogActionProvider.prototype, 'log');
-  }
+  beforeEach(() => (stub = createConsoleStub()));
+  beforeEach(async () => (logger = await getLogger()));
 
-  function restoreLogSpy() {
-    spy.restore();
-  }
+  afterEach(() => restoreLogSpy(stub));
 
-  function timer(startTime?: HighResTime): HighResTime {
-    if (!startTime) return [3, 3];
-    else return [0, 100000002];
-  }
+  it('logs a value without a start time', async () => {
+    const match = chalk.red('ERROR: /test :: TestClass.test() => test message');
+
+    await logger(req, [], 'test message');
+    sinon.assert.calledWith(stub, match);
+  });
+
+  it('logs a value with a start time', async () => {
+    const match = chalk.red(
+      'ERROR: 100ms: /test :: TestClass.test() => test message',
+    );
+    const startTime: HighResTime = logger.startTimer();
+
+    await logger(req, [], 'test message', startTime);
+    sinon.assert.calledWith(stub, match);
+  });
+
+  it('logs a value with args present', async () => {
+    const match = chalk.red(
+      'ERROR: /test :: TestClass.test(test, message) => test message',
+    );
+
+    await logger(req, ['test', 'message'], 'test message');
+    sinon.assert.calledWith(stub, match);
+  });
 });
+
+async function getLogger(logWriter?: LogWriterFn) {
+  class TestClass {
+    @log(LOG_LEVEL.ERROR)
+    test() {}
+  }
+
+  const provider = new LogActionProvider(
+    () => Promise.resolve(TestClass),
+    () => Promise.resolve('test'),
+    timer,
+  );
+
+  provider.logLevel = LOG_LEVEL.WARN;
```

Comment:
Isn't this line redundant, considering that `logLevel` is set to `WARN` by default?